### ISSUE TITLE
Can't seem to wrap carousel in flex box

### ISSUE THREAD

**Issue by realesh** [akiran/react-slick#1320]
I'm new here, just wanted the Center Mode to show in my single page web, I have followed the getting started guide and add the CDN to my html file,
problem is i cant seem to wrap the carousel into my flexbox, just wanted to make it cleaner and style it

here is my flexbox container using styled-components
![image](https://user-images.githubusercontent.com/34760718/43153408-689abc4e-8f9b-11e8-8234-14a0fbed1bb5.png)

and my Carousel.js
![image](https://user-images.githubusercontent.com/34760718/43153463-92cd6e62-8f9b-11e8-9ac1-e22da6e5ed45.png)
![image](https://user-images.githubusercontent.com/34760718/43153492-a3f7a162-8f9b-11e8-900b-39d69be334aa.png)

**Comment by Harshmakadia** [comment]
Even I'm facing the same issue looks like the slider is not working when the parent has flex property. Not sure if there is the solution to this issue.

**Comment by fredcorn** [comment]
Right now I just set the container to be display: block.

**Comment by edwardfhsiao** [comment]
hope there is a solution to fix this

**Comment by Harshmakadia** [comment]
@edwardfhsiao - If you wrap the component inside the container it will work 
`<div class="container"><Slider Component/></div>`

**Comment by edwardfhsiao** [comment]
@Harshmakadia 
By ```component```, do you mean
 ```js
class CustomSlider extends Component {
render(
return( 

      <Slider {...settings}>...</div>

 )
}
...
class Index extends Component {
render(
return( 
  <div class="container">
      <CustomSlider Component/>
  </div>
 )
}
```
?

Cuz I've  tried just wrapped it in to a div, still does not work

go here https://codesandbox.io/s/ppwkk5l6xx and replace the code with below
```js
import React, {  Component} from "react";
import ReactDOM from "react-dom";
import Slider from "react-slick";
import "./index.css";

 class CenterMode extends Component {
  render() {
    const settings = {
      className: "center",
      centerMode: true,
      infinite: true,
      centerPadding: "10px",
      slidesToShow: 3,
      speed: 500,
      arrows: false
    };
    return (
      <div style={{width: '40%', margin:'0 auto'}}>
        <h2>Center Mode</h2>
        <div style={{display: 'flex'}}>
        <Slider {...settings}>
          <div>
            <h3>1</h3>
          </div>
          <div>
            <h3>2</h3>
          </div>
          <div>
            <h3>3</h3>
          </div>

        </Slider>
      </div>
      </div>
    );
  }
}
ReactDOM.render(<CenterMode />, document.getElementById("container"));

```

**Comment by Destructor1702** [comment]
I'm having the same issue, even if a far above parent is flex everything brokes

**Comment by HsuTing** [comment]
I have the same issue, and I think should we add `width` to the parentNode of `<div className="slick-track" />`?

The `width` of the child node will effect the layout of `flex` .
However, the `width` of `<div className="slick-track" />` is the total width of the children node, but we just need one of the children node\`s `width`.
Maybe add a width to the parentNode of `<div className="slick-track" />` can fix this?

Do I make the sense?

**Comment by HsuTing** [comment]
I overwrite the width of `<div class='slick-track' />` and `<div class='slick-slide' />` with below patch:

```js
patch-package
--- a/node_modules/react-slick/lib/track.js
+++ b/node_modules/react-slick/lib/track.js
@@ -113,7 +113,9 @@ var renderSlides = function renderSlides(spec) {
       className: (0, _classnames2.default)(slideClasses, slideClass),
       tabIndex: "-1",
       "aria-hidden": !slideClasses["slick-active"],
-      style: _extends({ outline: "none" }, child.props.style || {}, childStyle),
+      style: _extends({ outline: "none" }, child.props.style || {}, childStyle, {
+        width: `calc(100% / ${spec.children.length * 2 + 1})`,
+      }),
       onClick: function onClick(e) {
         child.props && child.props.onClick && child.props.onClick(e);
         if (spec.focusOnSelect) {
@@ -137,7 +139,9 @@ var renderSlides = function renderSlides(spec) {
           tabIndex: "-1",
           className: (0, _classnames2.default)(slideClasses, slideClass),
           "aria-hidden": !slideClasses["slick-active"],
-          style: _extends({}, child.props.style || {}, childStyle),
+          style: _extends({}, child.props.style || {}, childStyle, {
+            width: `calc(100% / ${spec.children.length * 2 + 1})`,
+          }),
           onClick: function onClick(e) {
             child.props && child.props.onClick && child.props.onClick(e);
             if (spec.focusOnSelect) {
@@ -159,7 +163,9 @@ var renderSlides = function renderSlides(spec) {
           tabIndex: "-1",
           className: (0, _classnames2.default)(slideClasses, slideClass),
           "aria-hidden": !slideClasses["slick-active"],
-          style: _extends({}, child.props.style || {}, childStyle),
+          style: _extends({}, child.props.style || {}, childStyle, {
+            width: `calc(100% / ${spec.children.length * 2 + 1})`,
+          }),
           onClick: function onClick(e) {
             child.props && child.props.onClick && child.props.onClick(e);
             if (spec.focusOnSelect) {
@@ -192,14 +198,15 @@ var Track = exports.Track = function (_React$PureComponent) {
     var _props = this.props,
         onMouseEnter = _props.onMouseEnter,
         onMouseOver = _props.onMouseOver,
-        onMouseLeave = _props.onMouseLeave;
+        onMouseLeave = _props.onMouseLeave,
+        trackStyle = _props.trackStyle;

     var mouseEvents = { onMouseEnter: onMouseEnter, onMouseOver: onMouseOver, onMouseLeave: onMouseLeave };
     return _react2.default.createElement(
       "div",
       _extends({
         className: "slick-track",
-        style: this.props.trackStyle
+        style: _extends({}, trackStyle, { width: `calc(100% * ${slides.length})` }),
       }, mouseEvents),
       slides
     );
```

This work under `flex` to me.
Hope to help you.

**Comment by Gabriel-diez** [comment]
Setting { min-height: 0px; min-width: 0px; } on container fixed the issue for me.

**Comment by aclowwwn** [comment]
Having the same issue here, any solution would be highly appreciated. Thanks

**Comment by durdenx** [comment]
> Setting { min-height: 0px; min-width: 0px; } on container fixed the issue for me.
Thanks ! I've been looking for three hours....

**Comment by im-amir** [comment]
Anyone still having this issue, just put
* { min-height: 0; min-width: 0; }
in the css file

**Comment by MarkPare** [comment]
I'm still having this issue and above solutions didn't work.  Is it possible to use react slick when it has an ancestor that is `display: flex`?

**Comment by Vaisakhkm2625** [comment]
I fixed it 

> I'm still having this issue and above solutions didn't work. Is it possible to use react slick when it has an ancestor that is `display: flex`?

I also had same error and was looking for solution for a week now.
but i found out that the property is getting there through some other class due to improper naming of class and namespaces.

Hope in future everyone reading this will double check every classes and use modules....

**Comment by hitchcliff** [comment]
i fixed it by setting `height: 100%; width: 100%;`

**Comment by IsidoroAros** [comment]
Late response but you can fix it by setting a wrapper for the slider 

```
<div className='flex-container'>
  <> any other content </>
    **<div className='wrapper'>**
         <Slider {...props} />
    **<div />**
<div />
```

and your wrapper should have display: block; and width and height setted to 100%

**Comment by mshahzaib101** [comment]
for anyone facing this issue on slider child then try to wrap the child slide under the div like below:
<img width="583" alt="Screenshot 2023-12-26 at 2 15 12 PM" src="https://github.com/akiran/react-slick/assets/55647177/fac03f5d-564b-467b-9416-d157e248923e">

**Comment by ozzytop** [comment]
I see this issue... is there a solution for it?

**Comment by cucharachaa** [comment]
Same issue here =(

**Comment by ninajhun** [comment]
TL;DR: If you are using **grid**, you cannot use fractional units `fr`. The container needs to have a defined width.

Had this issue when trying to use the Slider in a `display:grid` layout. The workaround that most people seem to be suggesting is to add some sizing classes to the container.  For me, all  I needed was `width` while setting `{ min-height: 0; min-width: 0; }` directly on the container didn't work.
```

	<div style={{ width: '600px' }}> // container needs a defined width 
		<Slider>
              ...rest of code
		</Slider>
	</div>
```

HOWEVER, I found the source of the issue in the grid for myself! The `grid-template-columns` cannot be set to `fr` units! I think the carousel needs a set width hence why add the width to the container classes did the trick before. 

```
.your-grid-layout {
	display: grid;
	grid-template-columns: 50% 50%;   // 1fr 1fr would cause the flickering issue here
	...rest of code
}
```
